From 523b347ad49234efb7787f6f084070034de7e977 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Wed, 27 May 2020 15:49:49 +0200
Subject: [PATCH] nllistener: increase netlink buffer to 64k

Currently 4k buffer is too small to handle some netlink messages

(Like bridge vlans for example, with 32k messages detected).
---
 ifupdown2/nlmanager/nllistener.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ifupdown2/nlmanager/nllistener.py b/ifupdown2/nlmanager/nllistener.py
index b40b058..e319dc4 100644
--- a/ifupdown2/nlmanager/nllistener.py
+++ b/ifupdown2/nlmanager/nllistener.py
@@ -44,7 +44,7 @@ class NetlinkListener(Thread):
     # As defined in asm/socket.h
     _SO_ATTACH_FILTER = 26
 
-    RECV_BUFFER = 4096  # 1024 * 1024
+    RECV_BUFFER = 65536  # 1024 * 1024
 
     def __init__(self, manager, groups, pid_offset=1, error_notification=False, rcvbuf_sz=10000000, bpf_filter=None):
         """
